pages/index.js

```diff
@@ -1,12 +1,28 @@
 import ExpirationOverview from "@/component/ExpirationOverview";
 import FilterButton from "@/component/FilterButton";
+import SearchBar from "@/component/SearchBar";
 import Link from "next/link";
+import { useState } from "react";
 
 export default function HomePage({ groceriesList }) {
+  const [filteredGroceriesList, setFilteredGroceriesList] =
+    useState(groceriesList);
+
+  const handleSearch = (searchTerm) => {
+    if (searchTerm.trim() === "") {
+      setFilteredGroceriesList(groceriesList);
+    } else {
+      const filteredList = groceriesList.filter((item) =>
+        item.name.toLowerCase().includes(searchTerm.toLowerCase())
+      );
+      setFilteredGroceriesList(filteredList);
+    }
+  };
```

Comment:
Good search function! You might want to convert your searchTerm to lowercase before for better readability. Also since `if (searchTerm.trim() === ""` should check if the search bar is empty, you could just check truthy falsy value instead and implement an early return.

```suggestion
const handleSearch = (searchTerm) => {
  const term = searchTerm.trim().toLowerCase();

  if (!term) {
    setFilteredGroceriesList(groceriesList);
    return;
  }

  const filteredList = groceriesList.filter(
    (item) => item.name.toLowerCase().includes(term)
  );

  setFilteredGroceriesList(filteredList);
};
```